logdna/data_source_alert.go

```diff
@@ -26,6 +26,13 @@ var alertProps = map[string]*schema.Schema{
 	"triggerlimit":    intSchema,
 }
 
+var _ = registerTerraform(TerraformInfo{
```

Comment:
I imagine `registerTerraform` is returning an error or nil? If so, would it be better to add an init function and handle the error?

```go
func init() {
  err := registerTerraform(/* stuff */)
  if err != nil {
    panic(err)
  }
}
```

Reply:
Unfortunately we can't, all the terraform resources and data sources need to be registered before Terraform initializes the provider.

The init function in our plugin isn't called until after Terraform does their init function, and Terraform initializes the provider in their own init function. So we need to register the resources at a global level so it happens before Terraform initializes.

`registerTerraform` doesn't actually need to return anything. It's just returning something so I can assign it to an empty variable so I can execute it in the global space. Kind of hacky, but it works.

Reply:
🤔 That sounds like a good comment to add about why we are ignoring the error.